src/interceptors/exception.interceptor.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  ArgumentsHost,
+  Catch,
+  ExceptionFilter,
+  HttpException,
+  Logger,
+} from '@nestjs/common';
+import { Response } from 'express';
+import { APIResponse } from 'src/utils/apiResponse.utils';
+
+@Catch(HttpException)
+export class HttpExceptionFilter implements ExceptionFilter {
+  private logger = new Logger();
```

Comment:
PrimathonAICodeReviewer

Consider adding a comment to describe the purpose of the `@Catch` decorator and the `HttpExceptionFilter` class.
```diff
+ // Decorator to catch HTTP exceptions and handle them
```



<!-- This is an auto-generated comment by OSS PrimathonAICodeReviewer -->